control_panel_api/tests/test_services.py

```diff
@@ -0,0 +1,96 @@
+        self.assertIn('UpdateRenameAndDeleteObjects', sids)
+
+        document = services.get_policy_document('foo', readwrite=False)
+
+        sids = [s['Sid'] for s in document['Statement']]
+        self.assertNotIn('UpdateRenameAndDeleteObjects', sids)
+
+    @patch('control_panel_api.aws.put_bucket_logging')
+    @patch('control_panel_api.aws.create_bucket')
+    def test_create_bucket(self, mock_create_bucket, mock_put_bucket_logging):
+        services.create_bucket('bucketname')
+
+        mock_create_bucket.assert_called_with('test-bucketname', region='eu-test-2', acl='private')
+        mock_put_bucket_logging.assert_called_with('test-bucketname', target_bucket='moj-test-logs',
+                                                   target_prefix='test-bucketname/')
+
+    @patch('control_panel_api.aws.delete_bucket')
+    def test_delete_bucket(self, mock_delete_bucket):
+        services.delete_bucket('bucketname')
+
+        mock_delete_bucket.assert_called_with('test-bucketname')
+
+    @patch('control_panel_api.aws.create_policy')
+    def test_create_bucket_policies(self, mock_create_policy):
+        services.create_bucket_policies('bucketname')
+
+        expected_calls = [
+            call('test-bucketname-readwrite', services.get_policy_document('test-bucketname', True)),
+            call('test-bucketname-readonly', services.get_policy_document('test-bucketname', False)),
+        ]
+        mock_create_policy.assert_has_calls(expected_calls)
+
+    @patch('control_panel_api.aws.delete_policy')
+    @patch('control_panel_api.aws.detach_policy_from_entities')
+    def test_delete_bucket_policies(self, mock_detach_policy_from_entities, mock_delete_policy):
+        services.delete_bucket_policies('bucketname')
+
+        expected_calls = [
+            call('arn:aws:iam::1337:policy/test-bucketname-readwrite'),
+            call('arn:aws:iam::1337:policy/test-bucketname-readonly')
+        ]
+        mock_detach_policy_from_entities.assert_has_calls(expected_calls)
+
+        expected_calls = [
+            call('arn:aws:iam::1337:policy/test-bucketname-readwrite'),
+            call('arn:aws:iam::1337:policy/test-bucketname-readonly')
+        ]
+        mock_delete_policy.assert_has_calls(expected_calls)
+
+    @patch('control_panel_api.services.create_bucket')
+    @patch('control_panel_api.services.create_bucket_policies')
+    def test_bucket_create(self, mock_create_bucket_policies, mock_create_bucket):
+        services.bucket_create('bucketname')
+
+        mock_create_bucket_policies.assert_called()
+        mock_create_bucket.assert_called()
+
+    @patch('control_panel_api.services.delete_bucket_policies')
+    def test_bucket_create(self, mock_delete_bucket_policies):
+        services.bucket_delete('bucketname')
+
+        mock_delete_bucket_policies.assert_called()
+
+
+@override_settings(ENV='test', IAM_ARN_BASE='arn:aws:iam::1337')
+class NamingTestCase(SimpleTestCase):
+    def test_bucket_name_has_env(self):
```

Comment:
As discussed in Slack yesterday and also as per other comment, the user now provides the actual S3 bucket name including the env prefix, so we don't need to prefix this anymore.

Basically UI will make this request:

```
POST /s3buckets

{
  "name": "dev-bucketname"
}
```

So we don't need to alter it (yay :) )

Reply:
Ok but I would rather do that in another branch of work I want to get this merged first

Reply:
Fair enough, see if remove the env is too involved otherwise don't worry 👍